engines/python/setup/djl_python/request_output.py

```diff
@@ -0,0 +1,132 @@
+from dataclasses import field, dataclass
+from typing import List, Optional, Callable, Any, Dict, Union
+
+from djl_python.rolling_batch.rolling_batch import Token
+
+
+@dataclass
+class Sequence:
+    """Dataclass to store the generated tokens details of all the sequences in the request.
+    Attributes:
+        tokens: generated tokens of the sequence.
+        top_tokens: top tokens of the sequence.
+        cumulative_log_prob: cumulative log probability of the sequence.
+        finish_reason: finish reason of the sequence.
+        stop_reason: stop reason of the sequence.
+        generated_text: generated text of the sequence.
+    """
+    tokens: List[Token] = field(default_factory=lambda: [])
+    top_tokens: Optional[List[List[Token]]] = field(default_factory=lambda: [])
+    cumulative_log_prob: float = 0.0
+    finish_reason: str = False
+    stop_reason: Optional[str] = None
+    generated_text: str = ""
+
+    def set_next_token(self, token: Token):
+        self.tokens.append(token)
+        self.generated_text += token.text
+
+    def set_next_top_tokens(self, top_tokens: List[Token]):
+        self.top_tokens.append(top_tokens)
+
+    @property
+    def next_token(self):
+        return self.tokens[-1]
+
+    @property
+    def next_top_tokens(self):
+        return self.top_tokens[-1] if self.top_tokens else None
+
+    def tokens_as_dict(self):
+        return [token.as_dict() for token in self.tokens]
+
+    def top_tokens_as_dict(self):
+        top_tokens = []
+        for top_token in self.top_tokens:
+            top_tokens.append([token.as_dict() for token in top_token])
+        return top_tokens
+
+
+@dataclass
+class RequestInput:
+    """Base class for all request inputs.
+    Attributes:
+        request_id: The request ID.
+        output_formatter: Output formatter of the request
+        parameters: parameters in the request payload
+    """
+    request_id: int
+    output_formatter: Union[Callable, str] = None
+    parameters: Dict = field(default_factory=lambda: [])
```

Comment:
not only parameters, we might want to keep the whole json here

Reply:
I agree, but the logic to get the whole json is in parse_input https://github.com/deepjavalibrary/djl-serving/blob/master/engines/python/setup/djl_python/utils.py#L94
We only parse "inputs" and "params",  only for chat completion, we take the whole json and parsed into ChatProperties. 

And this params is passed down to rolling batch inference.

Reply:
then it is better to register a new request there if possible. SF customer need some custom field from the json other than parameter itself

Reply:
Sure, will do this in the next PRs.